File: python/clock/clock.py
class Clock(object):
    def __init__(self, hours, minutes):
        self.hours = hours
        self.minutes = minutes

    def __str__(self):
        total_hours = self.hours + int(self.minutes//60)
        shift_hours = total_hours % 24
        shift_minutes = self.minutes % 60
        return "{h:02d}:{m:02d}".format(h=shift_hours, m=shift_minutes)

    def add(self, minutes):
        self.minutes += minutes
        return self

    def __eq__(self, other):
        return str(self) == str(other)

File: clojure/clock/src/clock.clj
(ns clock)


(defn- norm [m] (mod m 1440))


(defn clock->string [clock]
  (format "%02d:%02d" (quot clock 60) (mod clock 60)))


(defn clock [hours minutes] (norm (+ (* 60 hours) minutes)))


(defn add-time [clock time] (norm (+ clock time)))
